client/src/helpers/unwrapCourse.jsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from "@mui/material";
+import React from "react";
+
+const COURSE_CODE_PATTERN = /([A-Z]{2,4}\s[0-9]{3})/g;
+const START_MARKER_PATTERN = /\$\[/g;
+const END_MARKER_PATTERN = /\]\$/g;
+
+/**
+ * Replaces course codes with searchable buttons, and remove the markings
+ *
+ * @param {course} marked string representing prerequisites or dependencies
+ * @param {courseHandler} courseHandler function used to search for a course when clicked on it
+ * @returns an unmarked string representing prerequisites or dependencies, where courses are replaced
+ *        with re-searchable buttons
+ */
+export function unwrapCourse(course, courseHandler) {
+  // create an array from the string, where where even-numbered elements are non-course code parts,
```

Comment:
Maybe it is just  because I am unfamiliar with React, but I am not sure why course codes would all be even numbered indices

Reply:
Great question!

It's actually a JS thing. The .split() method I use, while passing in a regular expression, splits the string into an array but the pattern matched in the regular expression. But, it does not merely store the pattern matches, rather the parts of the string that are non-matching--exactly what we want! The way it does this is by storing the pattern-matched substrings at odd indices and the remainder substrings at even indices.